### PR TITLE
expose host_emptyCache to python, fix a bug in freeing cudaHostRegist…

### PR DESCRIPTION
**Pull request by ngimel**
…ered memory

Fixes #ISSUE_NUMBER

### CODE REVIEW

File: aten/src/ATen/cuda/CachingHostAllocator.cpp [hunk 1]
```diff
@@ -98,7 +98,8 @@ struct CUDACachingHostAllocatorImpl
             pinned_use_cuda_host_register()) {
       void* ptr = block->ptr_;
       AT_CUDA_CHECK(cudaHostUnregister(ptr));
-      free(ptr);
+      // call C-library free instead of ours (that frees a block)
+      ::free(ptr);
```
Comment (eqy) [inline aten/src/ATen/cuda/CachingHostAllocator.cpp:102]: Is `std::free` more explicit here or is that incorrect?

**Review by eqy — APPROVED** [review]

File: aten/src/ATen/cuda/CachingHostAllocator.cpp (hunk 1 above)
Comment (ngimel) [reply aten/src/ATen/cuda/CachingHostAllocator.cpp:102]: YEah std::free should work too (and we should replace corresponding malloc with std::malloc then)

**Comment by awgu** [comment]
should be able to close https://github.com/pytorch/pytorch/issues/134332 after this iiuc

**Comment by ngimel** [comment]
@albanD I'll let you guys decide where the proper python wrapper should live and what the name of the function should be, at least it can be called now, and the bug in empty_cache is fixed.

**Pull request closed without merging**